File: environment.py
class Environment:
    def __init__(self):
        self.numVariables = 0
        self.variables = None
        self.parent = None
        self.child = None
        
    def getNumVariables(self):
        return self.numVariables
    def getVariables(self):
        return self.variables
    def getParetn(self):
        return self.parent
    def getChild(self):
        return self.child

File: environment.c
#include "environment.h"

#include <stdlib.h>

/**
  * environment.c
  *
  * Authors: Adan Nunez, Garo Anguiano-Sainz, Colby Seyferth
  * Date: March 05, 2014
  * CS 251: Programming Languages
  *
  * Implements environment.h
  *
  **/

void freeEnvironments(Environment *env){
	if (env == NULL){
		return;
	}
	else if (env->child != NULL){
		freeEnvironments(env->child);
		env->child = NULL;
	}
	free(env->variables);
	free(env);
}